Source/WebCore/page/Page.cpp

```diff
@@ -1926,7 +1926,8 @@ void Page::updateRendering()
     });
 
     runProcessingStep(RenderingUpdateStep::Reveal, [] (Document& document) {
-        document.reveal();
+        if (document.visibilityState() != VisibilityState::Hidden)
+            document.reveal();
```

Comment:
Maybe have a follow up to figure out why  this check isn't part of `forRenderableDocument`